web/src/scheduled_messages_ui.js

```diff
@@ -102,3 +105,54 @@ export function initialize() {
         e.stopPropagation();
     });
 }
+
+export function should_show_schedule_message_indicator() {
+    const scheduled_messages_list = Object.values(scheduled_messages.scheduled_messages_data);
+    const filter = narrow_state.filter();
+    const is_conversation_view = filter === undefined ? false : filter.is_conversation_view();
+    const current_narrow = narrow_state.narrowed_to_pms() ? "private" : "stream";
+    if (!is_conversation_view) {
+        return false;
+    }
+    const matching_scheduled_messages = scheduled_messages_list.filter((scheduled_message) => {
+        if (current_narrow === scheduled_message.type) {
```

Comment:
This would be a lot nicer to read with the early return pattern; return if `if current_narrow !== scheduled_message.type` and then you don't end up with all the code indented super deeply.

I might also consider doing a normal if/else chain instead of the `switch` statement.

Reply:
Ah yeah, that sounds better to me. I was quite tunnel visioned on getting the logic down when I was working on it, now it looks better.